src/main/java/com/worldwarofants/game/util/RandGen.java

```diff
@@ -0,0 +1,77 @@
+package com.worldwarofants.game.util;
+
+import java.util.Random;
+
+/*
+    Utility class that provides an abstraction of Java's Random class.
+    Its goal is to provide developers with a simple way to work with
+    random number generation, as well as allow for precise tweaking
+    of exact methods used.
+
+    Please make sure to generate your random numbers within the project
+    exclusively using this package.
+*/
+
+public class RandGen extends java.util.Random {
+    private static final long serialVersionUID = -5378246343912387722L;
+
+    Random generator;
+    long seed;
+    String seedString; // If a custom text input has been provided by user and hashed,
+
+    // Overloaded seedless constructor
+    public RandGen() {
```

Comment:
There is probably a way you can use constructor chaining here. Read about it online. In short, you use one constructor inside another constructor, so you do not repeat the code. Here, you could generate some string or whatever, and  you can just call RandGen(seedString), and you will save yourself all of these lines you're repeating.

By the way, don't be stingy with the naming. There's no problem calling the class RandomGeneration, even though that's not a very descriptive name either. SeededGeneration or something like that is closer to what is happening here. Anyway, longer names are not a problem because most IDEs have auto completion anyway. And it is more understandable.

Reply:
I am not sure it would be possible to chain these together, as they both perform different operations from the first line.